dashboard/app/helpers/aichat_safety_helper.rb

```diff
@@ -24,16 +25,43 @@ def find_toxicity(role, text, locale)
         comprehend_response = AichatComprehendHelper.get_toxicity(text, locale)
         return {text: text, blocked_by: 'comprehend', details: comprehend_response} if comprehend_response && comprehend_response[:toxicity] > threshold
       end
+
+      if openai_enabled?(role)
+        details = openai_safety_check(text)
+        return {text: text, blocked_by: 'openai', details: details} if details
+      end
+    end
+
+    private def openai_safety_check(text)
+      details = nil
+      # Try twice in case of network errors or model not correctly following directions and
+      # replying with something other valid expected output.
+      Retryable.retryable(tries: 2) do
```

Comment:
For my understanding, does `Retryable` automatically retry if an error was raised in the block (i.e. we don't need to `rescue` it ourselves, right)?

Reply:
yes

Reply:
Added tests to confirm this.